### ISSUE TITLE
xz dependency

### ISSUE THREAD

**Issue by bovine3dom** [davidanthoff/NodeJS.jl#39]
`unxz` must be on the path to build this package. This isn't available on all systems (e.g. Docker's "official" julia image https://github.com/docker-library/julia).

Is there a way to automatically pull in this dependency (or swap it for one that can be) during the build process?

**Comment by davidanthoff** [comment]
On Julia 1.3 the package now uses the new artifacts system, so I think this shouldn’t be an issue there anymore. Could you try that?

**Comment by bovine3dom** [comment]
The issue is present on 1.3 - Dockerfile is here (includes a workaround) https://github.com/IntegratedTransportPlanning/gcvt/blob/master/src/backend/Dockerfile

xz isn't included in the artifacts. Hmm, maybe it's an upstream issue? I don't see any mention of xz in this repository at all.

Edit: hmm, I can't replicate it now with `sudo docker run -it julia:1.3 # then ] add NodeJS`. I'll do some more digging :)

**Comment by bovine3dom** [comment]
I get the error with my {Project,Manifest}.toml but not if I just add NodeJS.jl, so this must be an upstream issue. Thanks for your time and sorry for the noise!

**Comment by bovine3dom** [comment]
Aha, I found the issue - version 1.0 of NodeJS.jl was installed. It was installed as an indirect dependency of VegaLite. NodeJS.jl 1.1.1 works fine.

Might I suggest adding 1.1 to https://github.com/queryverse/VegaLite.jl/blob/2f0df458678400ba60a6d1ed85a101a731a52c95/Project.toml#L35 ?

**Comment by davidanthoff** [comment]
So specifying `1` there means that VegaLite.jl is compatible with all `1.x.y` versions of NodeJS, including 1.1.1. So I suspect that all that was needed was an `pkg> up` on your project, right?

**Comment by bovine3dom** [comment]
I'll give that a go today. I'm not sure the timeline quite makes sense as I added [VegaLite 6 days ago](https://github.com/IntegratedTransportPlanning/gcvt/commit/91cf2e2c36e94d716460ddf016a0720d6b3bea43), which is about a week after NodeJS.jl 1.1 was released. I don't understand why it picked NodeJS.jl 1.0.

Edit: yeah, `pkg> up` fixed it. I still can't quite wrap my head around why NodeJS.jl 1.0 was ever installed. Thanks for your time on this :)
